### DISCUSSION TITLE
Accelerometer stopped working after moving from Zephyr 2 to 3

### DISCUSSION THREAD

**Discussion by jefffhaynes** [zephyrproject-rtos/zephyr#50884] [Q&A]
Not sure how to ask for help with this because I'm super lost.  I have what I believe is a pretty straightforward use of an accelerometer.  It was working both on our dev board and our production board under Zephyr 2.  After moving to 3, we are getting EIO from the I2C bus when we try to contact the part.

Here is our overlay for the nrf52840 SDK:

```
&i2c0 {
	lis2ds12: lis2ds12@1d {
		compatible = "st,lis2ds12";
		reg = <0x1d>;
		irq-gpios = <&gpio0 0x1e GPIO_ACTIVE_HIGH>;
		odr = <2>;
		power-mode = <1>;
	};
};
```

And the relevant code:

```
   _device = DEVICE_DT_GET_ANY(st_lis2ds12);

    if (!device_is_ready(_device)) 
    {
        // returns false, device is not ready b/c of EIO
    }
```

The only difference with the 2 version was that I was retrieving the device by label, but I don't see how that would matter.

EDIT: There doesn't appear to be any I2C traffic on the bus at all.

@avisconti any ideas?

**Comment by yashi** [comment]
The doc says:

> If this returns non-NULL, the device must be checked for readiness before use, e.g. with [device_is_ready()](https://docs.zephyrproject.org/latest/kernel/drivers/index.html#group__device__model_1gaa4944bd850e90cbd52b0489f9b12edfb).

https://docs.zephyrproject.org/latest/kernel/drivers/index.html#c.DEVICE_DT_GET_ANY
